xbrowse_server/base/model_utils.py

```diff
@@ -0,0 +1,182 @@
+import logging
+#from xbrowse_server.base.models import Project as BaseProject, Family as BaseFamily, Individual as BaseIndividual
+from seqr.models import Project as SeqrProject, Family as SeqrFamily, Individual as SeqrIndividual, \
+    VariantTagType as SeqrVariantTagType, VariantTag as SeqrVariantTag, VariantNote as SeqrVariantNote
+
+XBROWSE_TO_SEQR_CLASS_MAPPING = {
+    "Project": SeqrProject,
+    "Family": SeqrFamily,
+    "Individual": SeqrIndividual,
+    "ProjectTag": SeqrVariantTagType,
+    "VariantTag": SeqrVariantTag,
+    "VariantNote": SeqrVariantNote,
+}
+
+XBROWSE_TO_SEQR_FIELD_MAPPING = {
+    "Project": {
+        "project_name": "name",
+        'project_id': 'deprecated_project_id',
+        'last_accessed_date': 'deprecated_last_accessed_date',
+    },
+    "Family": {
+        "family_name": "display_name",
+        "short_description": "description",
+        "about_family_content": "analysis_notes",
+        "analysis_summary_content": "analysis_summary",
+    },
+    "Individual": {
+        'gender': 'sex',
+        'nickname': 'display_name',
+        'phenotips_id': 'phenotips_eid',
+    },
+    "ProjectTag": {
+        "tag": "name",
+        "title": "description",
+    },
+    "VariantTag": {
+        "project_tag": "variant_tag_type",
+        "date_saved": "created_date",
+        "user": "created_by",
+        "xpos": "xpos_start",
+    },
+    "VariantNote": {
+        "project_tag": "variant_tag_type",
+        "date_saved": "created_date",
+        "user": "created_by",
+        "xpos": "xpos_start",
+    },
+}
+
+
+def _update_model(model_obj, **kwargs):
+    for field, value in kwargs.items():
+        setattr(model_obj, field, value)
+
+    model_obj.save()
+
+
+def _is_xbrowse_model(obj):
+    return type(obj).__module__ == "xbrowse_server.base.models"
+
+
+def find_matching_seqr_model(xbrowse_model):
+    if not _is_xbrowse_model(xbrowse_model):
+        raise ValueError("Unexpected model class: %s.%s" % (type(xbrowse_model).__module__, type(xbrowse_model).__name__))
+
+    try:
+        xbrowse_class_name = type(xbrowse_model).__name__
+
```

Comment:
Don't all the xbrowse models keep references to their seqr counterparts? For example, the xbrowse Project has a `seqr_project` field, the xbrowse Family has a `seqr_family` field, etc. Why not try those first?

Reply:
good point. I'll change this.

Reply:
fixed.